### PR TITLE
all: replace unnecesary fmt.Errorf("[^%]*") calls with errors.New

### PR DESCRIPTION
**Pull request by odeke-em**
After running
```shell
grep -Rn 'fmt.Errorf("[^%]*")' *
```
noticed that there were too many unnecessary function calls to
fmt.Errorf(PLAIN_STRING) of which these calls can be directly
replaced with errors.New(PLAIN_STRING), which is anyways invoked
by fmt.Errorf and this change avoids incurring extraneous
function calls at runtime.

This work is a token from your friends from @tendermint

### CODE REVIEW

File: eth/backend.go [hunk 1]
```diff
@@ -296,6 +296,10 @@ func (s *Ethereum) ResetWithGenesisBlock(gb *types.Block) {
 	s.blockchain.ResetWithGenesisBlock(gb)
 }
 
+var (
+	errEtherBaseAddressWanted = errors.New("etherbase address must be explicitly specified")
```
Comment (ebuchman) [inline eth/backend.go]: Etherbase (for consistency - "b" shouldn't be capitalized)

File: rlp/encode.go [hunk 2]
```diff
@@ -428,9 +429,13 @@ func writeBigIntNoPtr(val reflect.Value, w *encbuf) error {
 	return writeBigInt(&i, w)
 }
 
+var (
+	errRLPEncodeWantPositiveBigInt = errors.New("rlp: cannot encode negative *big.Int")
```
Comment (ebuchman) [inline rlp/encode.go]: technically should be `WantNonNegative`. Maybe `errRLPEncodeNegativeBigInt` is sufficient

File: swarm/network/syncer.go [hunk 3]
```diff
@@ -231,13 +232,17 @@ func encodeSync(state *syncState) (*json.RawMessage, error) {
 	return &meta, nil
 }
 
+var (
+	errFailedtoDeserialSyncStateFromNil = errors.New("unable to deserialise sync state from <nil>")
```
Comment (ebuchman) [inline swarm/network/syncer.go]: errDecodeNilSyncState

File: whisper/whisperv5/whisper.go [hunk 4]
```diff
@@ -377,7 +393,7 @@ func (w *Whisper) AddSymKeyFromPassword(password string) (string, error) {
 
 	// double check is necessary, because deriveKeyMaterial() is very slow
 	if w.symKeys[id] != nil {
-		return "", fmt.Errorf("critical error: failed to generate unique ID")
```
Comment (ebuchman) [inline whisper/whisperv5/whisper.go]: is this error really more "critical" than the previous ones?

File: eth/backend.go (hunk 1 above)
Comment (odeke-em) [inline eth/backend.go]: Gotcha, thanks, updating.

File: rlp/encode.go (hunk 2 above)
Comment (odeke-em) [inline rlp/encode.go]: Great clarity, thanks!

File: swarm/network/syncer.go (hunk 3 above)
Comment (odeke-em) [inline swarm/network/syncer.go]: Nice!

File: whisper/whisperv5/whisper.go (hunk 4 above)
Comment (odeke-em) [inline whisper/whisperv5/whisper.go]: Good question @ebuchman, I'll leave that to the Ethereum maintainers to answer as I don't have much context about the entire function.

**Comment by fjl** [comment]
I agree with @karalabe.

**Comment by odeke-em** [comment]
Thank you for the insight @karalabe, gotcha, makes sense and thanks for advocating for that. I'll update this soon.

**Pull request closed without merging**